python/plugins/processing/algs/gdal/DatasetIdentify.py

```diff
@@ -0,0 +1,132 @@
+***************************************************************************
+"""
+
+__author__ = "Even Rouault"
+__date__ = "December 2025"
+__copyright__ = "(C) 2025, Even Rouault"
+
+import os
+
+from qgis.core import (
+    Qgis,
+    QgsProcessing,
+    QgsProcessingException,
+    QgsProcessingParameterFile,
+    QgsProcessingParameterBoolean,
+    QgsProcessingParameterVectorDestination,
+)
+from processing.algs.gdal.GdalAlgorithm import GdalAlgorithm
+from processing.algs.gdal.GdalUtils import GdalUtils
+
+pluginPath = os.path.split(os.path.split(os.path.dirname(__file__))[0])[0]
+
+
+class DatasetIdentify(GdalAlgorithm):
+    INPUT = "INPUT"
+    RECURSIVE = "RECURSIVE"
+    DETAILS = "DETAILS"
+    OUTPUT = "OUTPUT"
+
+    def __init__(self):
+        super().__init__()
+
+    def initAlgorithm(self, config=None):
+        self.addParameter(
+            QgsProcessingParameterFile(
+                self.INPUT,
+                self.tr("Input folder"),
+                Qgis.ProcessingFileParameterBehavior.Folder,
+            )
+        )
+
+        self.addParameter(
+            QgsProcessingParameterBoolean(
+                self.RECURSIVE,
+                self.tr("Perform recursive exploration of the input folder"),
+                defaultValue=True,
+            )
+        )
+
+        detailsParam = QgsProcessingParameterBoolean(
+            self.DETAILS,
+            self.tr("Add details about identified datasets in the output"),
+            defaultValue=True,
+        )
+        detailsParam.setHelp(
+            self.tr(
+                "Reports if the file is a Cloud Optimized GeoTIFF (COG), the list of files that compose the dataset, if it has georeferencing (has_geotransform and has_crs fields) and overviews"
+            )
+        )
+        self.addParameter(detailsParam)
+
+        self.addParameter(
+            QgsProcessingParameterVectorDestination(
+                self.OUTPUT,
+                self.tr("Output file"),
+                QgsProcessing.SourceType.TypeVector,
+            )
+        )
+
+    def name(self):
+        return "dataset_identify"
+
+    def displayName(self):
+        return self.tr("Dataset identification")
+
+    def group(self):
+        return self.tr("Miscellaneous")
+
+    def groupId(self):
+        return "miscellaneous"
+
+    def commandName(self):
+        return "gdal dataset identify"
+
+    def getConsoleCommands(self, parameters, context, feedback, executing=True):
+        arguments = []
+        if self.parameterAsBoolean(parameters, self.RECURSIVE, context):
+            arguments.append("--recursive")
+        if self.parameterAsBoolean(parameters, self.DETAILS, context):
+            arguments.append("--detailed")
+
+        inDirectory = self.parameterAsFile(parameters, self.INPUT, context)
+        if not os.path.exists(inDirectory):
+            raise QgsProcessingException(
+                self.tr(f"Directory {inDirectory} does not exist")
```

Comment:
Unfortunately, f-strings does not work with translations, we should use `.format()` here.

```suggestion
                self.tr("Directory {} does not exist".format(inDirectory))
```

Reply:
I think you forgot to push?

Reply:
actually, I didn't forget, but I see that the pyupgrade step of pre-commit  auto-changed to a f-string... I've pre-commit uninstall temporarily this time to avoid this

Reply:
... but our pre-commit CI job complains: https://github.com/qgis/QGIS/actions/runs/20215743964/job/58028381970?pr=64249

Reply:
Hmm - there's other places we do this (eg https://github.com/qgis/QGIS/blob/master/python/plugins/processing/algs/gdal/ogr2ogr.py#L121 ). Let's just ignore the failing pre-commit job and merge anyway...

Reply:
I could have stared at those two lines for an hour and not spotted that :rofl: